wire-ios-sync-engine/Tests/Source/Mocks/MockActionHandler.swift

```diff
@@ -0,0 +1,47 @@
+//
+// Wire
+// Copyright (C) 2024 Wire Swiss GmbH
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see http://www.gnu.org/licenses/.
+//
+
+import Foundation
+
+class MockActionHandler<T: EntityAction>: EntityActionHandler {
```

Comment:
Was this copied from data model? If so, I think we should move it to the data model support framework and import it where needed.

Reply:
I copied it because it was already being copied to many places. And currently I didn't find it in any shared frameworks.

Reply:
let's avoid creating duplication and instead move the code into a support framework.

Reply:
In a separate PR we can unify the mockActionHandler and put it on a shared handler

Reply:
Done! @johnxnguyen